lib/rspec/core/formatters/snippet_extractor.rb

```diff
@@ -1,114 +1,133 @@
+RSpec::Support.require_rspec_core "source"
+
 module RSpec
   module Core
     module Formatters
-      # @api private
-      #
-      # Extracts code snippets by looking at the backtrace of the passed error
-      # and applies synax highlighting and line numbers using html.
+      # @private
       class SnippetExtractor
```

Comment:
Your renaming of `SnippetExtractor` to `HtmlSnippetExtractor` suggests that maybe we shouldn't add a new class with the name `SnippetExtractor` because there really isn't such thing as a generic one.  Maybe this class would be better named `TerminalSnippetExtractor`?